### PR TITLE
Yarin&Yotam - PLAYG-28 - capture website to spreadsheet

### PR DESCRIPTION
**Pull request by yarinTabashi**

### CODE REVIEW

File: data_loader/test/test_training_storer_logic.py [hunk 1]
```diff
@@ -2,30 +2,30 @@
 
 from data_loader.training_storer import get_values, insert_values, delete_values
 
-TEST_INSERT_SPREADSHEET_ID = '1KQ2nFO9oeDLhHmKh9UTvNnhepKhMuTOyKmh7Nnxbj80'
+TEST_SPREADSHEET_ID = '1KQ2nFO9oeDLhHmKh9UTvNnhepKhMuTOyKmh7Nnxbj80'
```
Comment (smartFreshBear) [inline data_loader/test/test_training_storer_logic.py:5]: Side quest: ask yourself, "why it is not risky" to put this ID here

File: training/test/test_iterator_and_storer.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        try:
+            while True:
+                reformatted = '\n'.join(result)
+                assert updated_spreadsheet[len(original_spreadsheet) + counter][0] == reformatted, "Spreadsheet content mismatch"
+                counter += 1
+                result = iterator.next()
+        except:
+            pass
+
+        training_storer.delete_values(cells_range='A6:A', target_spreadsheet=TEST_SPREADSHEET_ID)
```
Comment (smartFreshBear) [inline training/test/test_iterator_and_storer.py:48]: we want to do it anyway, whatever stormy day is occuring, so some where you should:
"try-catch-finally" see here example:
https://www.w3schools.com/python/ref_keyword_finally.asp

**Review by smartFreshBear — CHANGES_REQUESTED** [review]
it's starting look very good!
the only question mark above my head is that:
you developed WebsiteCrawler, I don't see test for it, 
rather, I see: 

training/test/test_iterator_and_storer.py

which test ALMOST all the flow of WebsiteCrawler, can you just keep it simple and test WebsiteCrawler?

File: training/test/test_iterator_and_storer.py (hunk 2 above)
Comment (yarinTabashi) [reply training/test/test_iterator_and_storer.py:48]: Sure! I will fix it.

File: training/test/test_website_crawler.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        updated_spreadsheet = training_storer.get_values(ignore_un_tagged=False, cells_range='A:A', target_spreadsheet=TEST_SPREADSHEET_ID)
+
+        # Suppose it's enough to compare the first rows and the last rows
+        assert len(updated_spreadsheet) == len(original_spreadsheet) + added_cells_count, "Spreadsheet length mismatch"
+        assert original_spreadsheet[0] == updated_spreadsheet[0] and original_spreadsheet[
+            len(original_spreadsheet)-1] == updated_spreadsheet[
+                   len(original_spreadsheet) - 1], "First rows or last rows don't match"
+
+        counter = 0
+        iterator = WebsiteIterator.get_iterator_for_website(self.test_link, caching_manager=self.temp_caching_manager, paragraph_size=DEFAULT_PARA_SIZE)
```
Comment (smartFreshBear) [inline training/test/test_website_crawler.py:35]: Why are we testing internal logic of the crawler here?  I don't see any apparent reason

Comment (yarinTabashi) [reply training/test/test_website_crawler.py:35]: This module should have been used as an integration test, which combines the **_website_iterator_** and **_training_storer_** modules. Although each one has a unit test, here we want to test the whole script without using mocks, unlike in a unit test. Specifically, we need to verify that when provided a URL, the data is correctly stored in the spreadsheet.
To accomplish this, we have to use an iterator to know the expected data to be seen in the updated spreadsheet.

I haven't found a better solution, but I would like to here your opinion.

**Review by smartFreshBear — APPROVED** [review]
Looks good! Approved!

**Pull request merged**